src/System.IO.Pipelines/System/IO/Pipelines/BufferEnumerator.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Buffers;
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// An enumerator over the <see cref="ReadableBuffer"/>
+    /// </summary>
+    public struct BufferEnumerator
```

Comment:
Why not nested type in ReadableBuffer?

Reply:
It's used in some other places where it's created using cursors only. I'll change it in next iteration.